bottlerocket/types/src/agent_config.rs

```diff
@@ -92,12 +92,15 @@ impl Configuration for MigrationConfig {}
 #[serde(rename_all = "camelCase")]
 pub struct EksClusterConfig {
     /// The name of the eks cluster to create or an existing cluster.
+    /// If an eksctl config is passed, the names must match.
     pub cluster_name: String,
 
     /// Whether this agent will create the cluster or not.
     pub creation_policy: Option<CreationPolicy>,
 
-    /// The AWS region to create the cluster. If no value is provided `us-west-2` will be used.
+    /// The AWS region to create the cluster. If no value is provided `us-west-2` will be used. If
+    /// an eksctl cluster is provided, this region should match the one specified as this is the
```

Comment:
This is slightly vague. Is "an eksctl cluster" referring to an existing EKS cluster, or an eksctl cluster config (the same messaging as the comment above for `cluster_name`?)?

This should be treated the same as `cluster_name` or anything that can specified in the eksctl cluster config since that should always have `region` included in there.